02/app.js

```diff
@@ -1,6 +1,42 @@
 const formEl = document.forms[0];
 formEl.addEventListener('submit', checkData);
 
+function checkData(e){
+    e.preventDefault();
```

Comment:
To rozwiązanie co zrobiłaś jest ok.
Jednak zastanawiam się czy nie prościej byłoby zapisywać tylko informacje o błędach.
Następnie "resetować"(czyścić) HTML dot. błędów i ponownie je wyświetlać - te aktualne.